superset-frontend/src/SqlLab/components/AceEditorWrapper.tsx

```diff
@@ -34,41 +33,44 @@ import {
 
 const langTools = ace.acequire('ace/ext/language_tools');
 
-const propTypes = {
-  actions: PropTypes.object.isRequired,
-  onBlur: PropTypes.func,
-  sql: PropTypes.string.isRequired,
-  schemas: PropTypes.array,
-  tables: PropTypes.array,
-  functionNames: PropTypes.array,
-  extendedTables: PropTypes.array,
-  queryEditor: PropTypes.object.isRequired,
-  height: PropTypes.string,
-  hotkeys: PropTypes.arrayOf(
-    PropTypes.shape({
-      key: PropTypes.string.isRequired,
-      descr: PropTypes.string.isRequired,
-      func: PropTypes.func.isRequired,
-    }),
-  ),
-  onChange: PropTypes.func,
+type HotKey = {
+  key: string;
+  descr: string;
+  name: string;
+  func: () => void;
 };
 
-const defaultProps = {
-  onBlur: () => {},
-  onChange: () => {},
-  schemas: [],
-  tables: [],
-  functionNames: [],
-  extendedTables: [],
-};
```

Comment:
i think you can set these default props on line 68 in the new file with something like:
```typescript
constructor({ onBlur = () => {}, onChange = () => {}, ... , ...restProps}: Props) {
  super({onBlur, onChange, ..., ...restProps});
```

then you don't need to add in all the checks like ` if (this.props.onBlur)`

Reply:
Ah nice, I like that pattern. I went on a bit of a fruitless rabbit hole a while back trying to figure out how to make defaultProps work with the Props types.

Reply:
actually, looks like this got fixed in typescript 3.

https://stackoverflow.com/a/37282264